src/SSA_stepper.jl

```diff
@@ -139,6 +141,11 @@ end
 DiffEqBase.add_tstop!(integrator::SSAIntegrator,tstop) = integrator.tstop = tstop
 
 function DiffEqBase.step!(integrator::SSAIntegrator)
+    integrator.tprev = integrator.t
+    if integrator.uprev != integrator.u
+        copy!(integrator.uprev, integrator.u)
```

Comment:
Yeah, I was just worried about the cost of this operation, since it's O(n) and would then effect some O(1) methods.

Reply:
I see. Do you have a suggestion for what to do about this?

Reply:
I wonder if we can set `uprev` to be a reference to the most recently saved solution value. The behaviour will only be fully correct if `save_everystep` is `true` however...

Reply:
Oh yeah, that's definitely an issue we want to avoid. This would have to be updated analogous to `u` being updated within the SSA. I guess it could be done by storing the preceding jump index in the aggregator, and then applying it to `uprev` when `u` is updated with the current jump through `affect!`. I don't think this update is usually rate limiting (maybe for `Direct` if there are very few jumps).

Reply:
Did you want to use `uprev` for something?

Reply:
I would like to compute the probability density of the generated trajectory alongside the SSA. Thus in a callback I need to check which reaction just happened. I do not actually need `uprev` but would like to be able to access either the index of the most recently selected reaction or the jump vector `du=u-uprev`.

Reply:
Maybe a better idea for my use case is to record the index of the most recently performed jump.

Reply:
We could add a `prev_jump` field to the aggregators to store this. I think you should be able to then access this through the integrator, but I need to double check what everything stores...

Reply:
This is where the state gets updated and we could then save the old value:

https://github.com/SciML/DiffEqJump.jl/blob/80fdaa6e6abffdfccb1a7624ddd69c83c4cb2b82/src/aggregators/ssajump.jl#L115

Reply:
(By old value I mean the index of the jump that was just executed.)

Reply:
That sounds very reasonable. I still would like to be able to access tprev, so I‘ll change the PR to only include that part.

Thank you!

Reply:
@ChrisRackauckas Can correct me if I've got this wrong, but it appears you can access the underlying JumpAggregation that would store such info through ` integrator.cb.affect!`. I can add a PR for that in the next day or so. We just need to update each of the eight SSAs in the aggregator folder so that their `JumpAggregation` has the `prev_jump` field, and then set that field within `update_state!`.

Reply:
yeah that should work